src/pages/settings/Wallet/PaymentMethodList.tsx

```diff
@@ -154,128 +140,120 @@ function dismissError(item) {
+    const defaultablePaymentMethodCount = filteredPaymentMethods.filter(
         (method) => method.accountType === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT || method.accountType === CONST.PAYMENT_METHODS.DEBIT_CARD,
     ).length;
     return defaultablePaymentMethodCount > 1;
 }
 
-/**
- * @param {String} actionPaymentMethodType
- * @param {String|Number} activePaymentMethodID
- * @param {String} paymentMethod
- * @return {Boolean}
- */
-function isPaymentMethodActive(actionPaymentMethodType, activePaymentMethodID, paymentMethod) {
+function isPaymentMethodActive(actionPaymentMethodType: string, activePaymentMethodID: string | number, paymentMethod: PaymentMethod) {
     return paymentMethod.accountType === actionPaymentMethodType && paymentMethod.methodID === activePaymentMethodID;
 }
 
-/**
- * @param {Object} item
- * @returns {String}
- */
-function keyExtractor(item) {
-    return item.key;
+function keyExtractor(item: PaymentMethod) {
+    return item.key ?? '';
 }
 
 function PaymentMethodList({
-    actionPaymentMethodType,
-    activePaymentMethodID,
-    bankAccountList,
-    buttonRef,
-    cardList,
-    fundList,
-    filterType,
-    isLoadingPaymentMethods,
-    onPress,
-    shouldShowSelectedState,
-    shouldShowAddPaymentMethodButton,
-    shouldShowAddBankAccount,
-    shouldShowEmptyListMessage,
-    shouldShowAssignedCards,
-    selectedMethodID,
+    actionPaymentMethodType = '',
+    activePaymentMethodID = '',
+    bankAccountList = {},
+    buttonRef = () => {},
+    cardList = {},
+    fundList = {},
+    filterType = '',
     listHeaderComponent,
-    onListContentSizeChange,
-    shouldEnableScroll,
-    style,
-    listItemStyle,
-}) {
+    isLoadingPaymentMethods = true,
+    onPress,
+    shouldShowSelectedState = false,
+    shouldShowAddPaymentMethodButton = true,
+    shouldShowAddBankAccount = true,
+    shouldShowEmptyListMessage = true,
+    shouldShowAssignedCards = false,
+    selectedMethodID = '',
+    onListContentSizeChange = () => {},
+    shouldEnableScroll = true,
+    style = {},
+    listItemStyle = {},
+}: PaymentMethodListProps) {
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const {translate} = useLocalize();
     const {isOffline} = useNetwork();
 
     const filteredPaymentMethods = useMemo(() => {
         if (shouldShowAssignedCards) {
-            const assignedCards = _.chain(cardList)
+            const assignedCards = Object.values(cardList ?? {})
                 // Filter by physical, active cards associated with a domain
-                .filter((card) => !card.isVirtual && card.domainName && CONST.EXPENSIFY_CARD.ACTIVE_STATES.includes(card.state))
-                .sortBy((card) => !CardUtils.isExpensifyCard(card.cardID))
-                .value();
+                .filter((card) => !card.isVirtual && !!card.domainName && CONST.EXPENSIFY_CARD.ACTIVE_STATES.includes(card.state ?? 0))
+                .toSorted((card1, card2) => {
```

Comment:
Is this fully tested on all platforms? I am getting crash on iOS

https://stackoverflow.com/questions/76006398/tosorted-works-only-in-a-browser

<img width="328" alt="Screenshot 2024-02-21 at 5 27 00 PM" src="https://github.com/Expensify/App/assets/108292595/c78ded55-5b59-4c14-92bc-438b8b855454">

Reply:
Is it possible that this code path wasn't triggered during our tests?

Reply:
It's not a conflict; this method isn't available on iOS. I'm not sure why we didn't catch this.

Reply:
I'm fixing this right now

Reply:
https://github.com/Expensify/App/pull/37073

Reply:
Because using `sort` together with React can lead to very difficult to debug bugs (as it's mutating the original array). I've seen buggy code related to `sort` in the Expensify code.

Using `sort` in this specific case **would** work, though. Still, I'd personally prefer to have some linter rule completely banning the usage of `sort` inside React components.

Reply:
I closed the first PR [(see explanation)](https://github.com/Expensify/App/pull/34257#discussion_r1498987734).

Second PR, switching back to `_.sortBy`: https://github.com/Expensify/App/pull/37078